### PR TITLE
add script for autosde dialog for creation custom size volume

### PR DESCRIPTION
**Pull request by eilam20**
Add storage namespace with classes to approve volume create in autosde catalog item request.

related to:
manageiq-providers-autosde:SDE-737_add_autosde_catalog_auto

### CODE REVIEW

**Comment by eilam20** [comment]
**manageiq-providers-autosde:SDE-737_add_autosde_catalog_auto:**
https://github.com/ManageIQ/manageiq-providers-autosde/pull/161

**Comment by agrare** [comment]
@eilam20 the spec failures look legitimate

**Comment by eilam20** [comment]
@agrare I ran the script clean_yaml_files to fix the yaml files.

**Comment by eilam20** [comment]
@agrare hi I saw that the tests looks ok, can you revue it?

**Comment by agrare** [comment]
@eilam20 I'd like to merge this together with https://github.com/ManageIQ/manageiq-providers-autosde/pull/161 in case you find any issues while testing changes there

File: content/automate/ManageIQ/Storage/Autosde/GetMethods/GetEms.class/__methods__/get_ems.rb
Comment (Fryguy) [inline content/automate/ManageIQ/Storage/Autosde/GetMethods/GetEms.class/__methods__/get_ems.rb:1]: debug stuff here

File: content/automate/ManageIQ/Storage/Autosde/Services/ServiceVolumeRequestApproval.class/__methods__/approve_request.rb
Comment (agrare) [inline content/automate/ManageIQ/Storage/Autosde/Services/ServiceVolumeRequestApproval.class/__methods__/approve_request.rb:21]: @eilam20 we should use the `create_volume_queue` method rather than using the autosde client directly

File: content/automate/ManageIQ/Storage/Autosde/Services/ServiceVolumeRequestApproval.class/__methods__/approve_request.rb
Comment (agrare) [inline content/automate/ManageIQ/Storage/Autosde/Services/ServiceVolumeRequestApproval.class/__methods__/approve_request.rb:22]: Does this belong in approve_request? I feel like this should be in its own state but not familiar with this @Fryguy ?

File: content/automate/ManageIQ/Storage/Autosde/Services/ServiceVolumeRequestApproval.class/__methods__/approve_request.rb
Comment (eilam20) [reply content/automate/ManageIQ/Storage/Autosde/Services/ServiceVolumeRequestApproval.class/__methods__/approve_request.rb:21]: Can you guide me how to reach this function throw the automation engine?

File: content/automate/ManageIQ/Storage/Autosde/Services/ServiceVolumeRequestApproval.class/__methods__/approve_request.rb
Comment (agrare) [inline content/automate/ManageIQ/Storage/Autosde/Services/ServiceVolumeRequestApproval.class/__methods__/approve_request.rb:4]: Minor, but `.where().first` should be `.find_by()` or in the case of `.find_by(:id => ...)` you can just use `.find($evm.root.attributes['dialog_ems'])`

**Review by agrare — APPROVED** [review]
Some comments but can be handled in a follow-up PR, will merge once the dependent automate-engine pr is merged

File: content/automate/ManageIQ/Storage/Autosde/Services/ServiceVolumeRequestApproval.class/__methods__/approve_request.rb
Comment (eilam20) [reply content/automate/ManageIQ/Storage/Autosde/Services/ServiceVolumeRequestApproval.class/__methods__/approve_request.rb]: Fixed it

**Comment by Fryguy** [comment]
Backported to `oparin` in commit 05e4b3b5556826397572b912dbb76488e986df7b.

```text
commit 05e4b3b5556826397572b912dbb76488e986df7b
Author: Adam Grare <adam@grare.com>
Date:   Fri Aug 26 09:15:38 2022 -0400

    Merge pull request #715 from Autosde/SDE-262253_service_catalog_automation
    
    add script for autosde dialog for creation custom size volume
    
    (cherry picked from commit 87d0a23c5bcc132ea47de39d4a62358315e0547a)
```

**Pull request merged**
